cmd/syncthing/cli/main.go

```diff
@@ -97,38 +97,41 @@ func Run() error {
 			operationCommand,
 			errorsCommand,
 			debugCommand,
+			{
+				Name:     "-",
+				HideHelp: true,
+				Usage:    "Read commands from stdin",
+				Action: func(ctx *cli.Context) error {
+					if ctx.NArg() > 0 {
+						return errors.New("command does not expect any arguments")
+					}
+
+					// Drop the `-` not to recurse into self.
+					args := make([]string, len(os.Args)-1)
+					copy(args, os.Args)
+
+					fmt.Println("Reading commands from stdin...", args)
+					scanner := bufio.NewScanner(os.Stdin)
+					for scanner.Scan() {
+						input, err := shlex.Split(scanner.Text())
+						if err != nil {
+							return errors.Wrap(err, "parsing input")
+						}
```

Comment:
This continue is kind of awkward. Made me pause for a second to figure out what the code was doing. Here's a suggestion:

Instead of the continue try this:

```go
if len(input) != 0 {
    err = app.Run(append(args, input...))
    if err != nil {
        return err
    }
}
```

Reply:
The continue is consistent with the style in use in Syncthing's code base: Keep things flat and return/continue/break early. Maybe it's a Go thing, maybe just "us" - in some other code-bases this seems to be frowned upon (only return/... at the end). I like it this way.